pkg/validation/app_test.go

```diff
@@ -675,10 +676,59 @@ func Test_ValidateApp(t *testing.T) {
 				newTestCatalog("giantswarm", "default"),
 			},
 		},
+		{
+			name: "application not allowed outside org namespace",
+			obj: v1alpha1.App{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "kiam",
+					Namespace: "org-eggs2",
+					Labels: map[string]string{
+						label.AppOperatorVersion: "2.6.0",
+					},
+				},
+				Spec: v1alpha1.AppSpec{
+					Catalog:   "giantswarm",
+					Name:      "kiam",
+					Namespace: "kube-system",
+					KubeConfig: v1alpha1.AppSpecKubeConfig{
+						InCluster: true,
+					},
+					Version: "1.4.0",
+				},
+			},
+			catalogs: []*v1alpha1.Catalog{
+				newTestCatalog("giantswarm", "default"),
+			},
+			expectedErr: "validation error: namespace kube-system is not allowed",
+		},
+		{
+			name: "application allowed in the org namespace",
+			obj: v1alpha1.App{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "kiam",
+					Namespace: "org-eggs2",
+					Labels: map[string]string{
+						label.AppOperatorVersion: "2.6.0",
+					},
+				},
+				Spec: v1alpha1.AppSpec{
+					Catalog:   "giantswarm",
+					Name:      "kiam",
+					Namespace: "org-eggs2",
+					KubeConfig: v1alpha1.AppSpecKubeConfig{
```

Comment:
Is there already a testcases for `in-cluster` false but a different namespace? That should still work, but having a test for it would be nice.

Reply:
Hmm, I do not see a test for that, I will add one

Reply:
What I have just said, I guess the first test case is exactly what we are looking for :D

Reply:
Namely the `flawless flow` with `inCluster: false`, `.spec.namespace: kube-system` nad `.metadata.namespace: eggs2`